src/DittoMachine.sol

```diff
@@ -175,8 +186,13 @@ contract DittoMachine is ERC721, ERC721TokenReceiver, ERC1155TokenReceiver {
         uint256 _tokenId,
         address _ERC20Contract,
         uint256 _amount,
-        bool floor
-    ) public returns (uint256) {
+        bool floor,
+        uint256 index // index at which to mint the clone
+    ) public returns (
+        uint256, // cloneId
+        uint256, // protoId
+        uint256  // index
```

Comment:
why return the index, it'll be the same as the argument passed.

Reply:
I guess if gas isn't an issue, it's ok to return it.

Reply:
honestly did this just to make testing easier for me, can change if you want

Reply:
yeah, I realised that after going through the tests. We can keep it as is.